lib/_stream_wrap.js

```diff
@@ -201,14 +201,13 @@ StreamWrap.prototype._dequeue = function dequeue(item) {
 };
 
 StreamWrap.prototype.doClose = function doClose(cb) {
-  const self = this;
-  const handle = self._handle;
+  const handle = this._handle;
 
-  setImmediate(function() {
-    while (self._list !== null) {
-      const item = self._list;
+  setImmediate(() => {
```

Comment:
this should be named as well, also can you make this function top level, so that the for loop under can be optimized.